### PR TITLE
Server basically complete

### PR DESCRIPTION
**Pull request by networking-sorceress**
The only thing left to do now is to implement the encryption functions, which will just be system function calls.

The client is going to be interesting. Backend-wise, it will be fairly simple. The front-end is the challenging part though. Haven't looked for python3 ncurses libs yet, but I bet they're just as cancer as the C ones.

### CODE REVIEW

File: src/server.py
Comment (Aqcurate) [inline src/server.py:45]: Manage the socket.error exception?

File: src/server.py
Comment (Aqcurate) [inline src/server.py:116]: Add a way to close the server connection? def exit(self):

File: main.py
Comment (Aqcurate) [inline main.py:18]: Wrap everything in if \_\_name__ = '\_\_main__' to make the code only run when it is executed as a program

File: src/server.py
Comment (EscapingGithub) [inline src/server.py]: Pick either snake case or camel case

File: src/server.py
Comment (Aqcurate) [inline src/server.py]: Is the system import ever used?

File: src/connection.py
Comment (Aqcurate) [inline src/connection.py:66]: Do you plan on dealing with data >4096 bytes?

File: src/queue.py
Comment (Aqcurate) [inline src/queue.py:17]: I'm assuming the Queue system is unfinished?

**Review by Aqcurate — DISMISSED** [review]
REDACTED

**Review by Aqcurate — COMMENTED** [review]
Alright, here are my pedant complaints. Feel free to ignore them all. Everything works as is.

Make sure you remove \_\_pycache__.

For future reference, variables should use snake_case, following similar method naming conventions. Semicolons should not be used. (It would be a pain to change that now though, so leave it as is.)

I'm going to assume the Queue system and encryption system is unimplemented, so I only gave it a cursory glance.

I'm not too too familiar with threading and sockets, so I'll leave it to you to handle exceptions.

I didn't get a chance to run the code, but it looks fine. Just look over my comments, and I'll merge if there are no further changes.

File: src/server.py
Comment (networking-sorceress) [reply src/server.py:45]: Actually, for this one, I want it to crash if it fails. That way the person who runs it knows there's already something running on the port.

File: src/connection.py
Comment (networking-sorceress) [reply src/connection.py:66]: Yes. In the client. By saying you can't send more than 2,000 characters, leaving the other 2k for metadata.

File: src/queue.py
Comment (networking-sorceress) [reply src/queue.py:17]: Pretty much. I am just using it as a way to organize threadsafe lists for now, I may implement some safer way of managing it later but for now good practice will do.

File: src/server.py
Comment (networking-sorceress) [reply src/server.py]: No, I was importing it at first because I thought I might. I'm gonna go through and remove all the unused imports.

**Comment by networking-sorceress** [comment]
Changes are made.

File: main.py
Comment (otakaran) [inline main.py:23]: Has the port been cleared? What are you going to do about cross-compatability, or are you just hoping for the best? Is there a way to automatically search and open a port on a server? Having conflicts here would not be good...

File: src/connection.py
Comment (otakaran) [inline src/connection.py:28]: Add 'exit'
_Assuming there would be no conflict_

File: src/connection.py
Comment (otakaran) [reply src/connection.py:66]: This seems like an unnecessary limitation... Could there be a convenient config option that admins can change. Also, what kind of message will the user receive if data > 2,000 chars?

File: src/server.py
Comment (otakaran) [reply src/server.py:45]: How would the user know the crash was caused by a port config problem instead os something else. At least give the user a print out error!

File: src/connection.py
Comment (networking-sorceress) [reply src/connection.py:66]: This is outside of the scope of this project. This is strictly the server side. Those messages will be handled on the client side.

**Review by Aqcurate — APPROVED** [review]
Alright, everything looks good!

**Pull request merged**
